app/queries/search_results_query.rb

```diff
@@ -7,30 +7,51 @@ class SearchResultsQuery
   CONDITION_ADWORDS_URL_CONTAINS = "array_to_string(adwords_top_urls, ', ') LIKE :word"
 
   def initialize(search_results = nil, filters = {})
-    @search_results = search_results || SearchResult.all
+    @search_results = search_results || SearchResult.order(:id).all
     @filters = filters
   end
 
   def call
     @search_results = filter_by_url(filters[:url_equals])
     @search_results = filter_adwords_url_contains(filters[:adwords_url_contains])
+    @search_results = filter_urls_contains_at_least(filters[:url_contains_at_least_one], 1)
+    @search_results = filter_urls_contains_at_least(filters[:url_contains_at_least_two], 2)
 
     @search_results
   end
 
   def filter_by_url(url)
     if url.present?
-      @search_results.where(CONDITION_URL_EQUALS, url: url)
+      @search_results.where(CONDITION_URL_EQUALS, url: url).order(:id)
     else
       @search_results
     end
   end
 
   def filter_adwords_url_contains(word)
     if word.present?
-      @search_results.where(CONDITION_ADWORDS_URL_CONTAINS, word: "%#{word}%")
+      @search_results.where(CONDITION_ADWORDS_URL_CONTAINS, word: "%#{word}%").order(:id)
     else
       @search_results
     end
   end
+
+  def filter_urls_contains_at_least(character, count = 1)
+    if character.present?
+      search_result_ids = @search_results.select do |search_result|
+        at_least_character_count?(search_result.adwords_top_urls, character, count) ||
+          at_least_character_count?(search_result.non_adwords_urls, character, count)
```

Comment:
Optional: This is easier to read IMO, but I'm fine if you keep the current one too 🙏 

```suggestion
        adwords_top_urls_contains = urls.any? { |url| url.count(character) >= count }
        non_adwords_urls_contains = urls.any? { |url| url.count(character) >= count }
        adwords_top_urls_contains || non_adwords_urls_contains
```

Reply:
@Tuubz I'd prefer to keep the current one 😄 So we would split to logic to filter URLs with matched character count.